src/Definition/Snowflake.php

```diff
@@ -70,6 +71,48 @@ public function toArray()
         ];
     }
 
+    /**
+     * @param array $options
+     * @return string|null
+     * @throws InvalidOptionException
+     */
+    private function processLength($options)
+    {
+        if (!isset($options['length'])) {
+            return null;
+        }
+        if (is_array($options['length'])) {
+            return $this->getLengthFromArray($options['length']);
+        }
+        return $options['length'];
+    }
+
+    /**
+     * @param array $lengthOptions
+     * @throws InvalidOptionException
+     * @return null|string
+     */
+    private function getLengthFromArray($lengthOptions)
+    {
+        $expectedOptions = ['character_maximum', 'numeric_precision', 'numeric_scale'];
+        $diff = array_diff(array_keys($lengthOptions), $expectedOptions);
+        if (count($diff) > 0) {
+            throw new InvalidOptionException(sprintf('Length option "%s" not supported', $diff[0]));
+        }
+
+        $characterMaximum = isset($lengthOptions['character_maximum']) ? $lengthOptions['character_maximum'] : null;
+        $numericPrecision = isset($lengthOptions['numeric_precision']) ? $lengthOptions['numeric_precision'] : null;
+        $numericScale = isset($lengthOptions['numeric_scale']) ? $lengthOptions['numeric_scale'] : null;
+
+        if (!is_null($characterMaximum)) {
+            return $characterMaximum;
+        }
+        if (!is_null($numericPrecision) && is_numeric($numericScale)) {
```

Comment:
nejakej duvod proc je tam jednou is_null a jednou is_numeric?

Reply:
hlubší význam to nemá - koukal jsem že to vrací taky null když tam nemá být